### PR TITLE
Update the `User` model to include the teleconsultation number

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/877/update-the-user-model-to-include-the-teleconsultation-number

### CODE REVIEW

**Comment by msasikanth** [comment]
Moving to WIP to fix a failing test

File: app/src/main/java/org/simple/clinic/user/LoggedInUserPayload.kt [hunk 1]
```diff
@@ -30,5 +30,8 @@ data class LoggedInUserPayload(
     val createdAt: Instant,
 
     @Json(name = "updated_at")
-    val updatedAt: Instant
+    val updatedAt: Instant,
+
+    @Json(name = "teleconsult_phone_number")
+    val teleconsultPhoneNumber: String?
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/user/LoggedInUserPayload.kt:36]: IIRC, we are also going to make other changes to the user model and add the "capabilities" property. Can we try to make this part of the same migration?

File: app/src/androidTest/java/org/simple/clinic/storage/migrations/Migration71AndroidTest.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package org.simple.clinic.storage.migrations
+
+import org.junit.Test
+import org.simple.clinic.assertColumns
+
+class Migration71AndroidTest : BaseDatabaseMigrationTest(70, 71) {
+
+  @Test
+  fun registered_facility_id_and_assigned_facility_id_should_be_added_to_patient_table() {
+    before.assertColumns("LoggedInUser", setOf(
```
Comment (vinaysshenoy) [inline app/src/androidTest/java/org/simple/clinic/storage/migrations/Migration71AndroidTest.kt:10]: Can we also create a default entry and run the migration to ensure that the values are retained after migration correctly?

File: app/src/main/java/org/simple/clinic/login/pin/LoginPinEffectHandler.kt [hunk 3]
```diff
@@ -88,7 +88,8 @@ class LoginPinEffectHandler @AssistedInject constructor(
         updatedAt = entry.updatedAt!!,
         loggedInStatus = User.LoggedInStatus.OTP_REQUESTED,
         registrationFacilityUuid = entry.registrationFacilityUuid!!,
-        currentFacilityUuid = entry.registrationFacilityUuid
+        currentFacilityUuid = entry.registrationFacilityUuid,
+        teleconsultPhoneNumber = null
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/login/pin/LoginPinEffectHandler.kt:92]: `OngoingLoginEntry` is created from the `UsersApi#activate` API endpoint which returns a complete `LoggedInUserPayload` which will include the teleconsultation number.

So setting it to `null` is the incorrect thing to do here since we should also update `OngoingLoginEntry` to save the teleconsult number.

File: app/src/main/java/org/simple/clinic/storage/migrations/Migration_71.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package org.simple.clinic.storage.migrations
+
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
+import org.simple.clinic.storage.inTransaction
+import javax.inject.Inject
+
+@Suppress("Classname")
+class Migration_71 @Inject constructor() : Migration(70, 71) {
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/storage/migrations/Migration_71.kt:9]: Just FYI, since we squash and merge now, we can move the migration to a different commit for easier review.

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/simple/clinic/user/LoggedInUserPayload.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/user/LoggedInUserPayload.kt:36]: Yes it has `capabilities` as well. But I wasn't sure it was confirmed and it was listed as a separate story. Should I go ahead with that or wait until all the API contracts are confirmed? I picked this up because it's the most simple API contract.

File: app/src/main/java/org/simple/clinic/storage/migrations/Migration_71.kt (hunk 4 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/storage/migrations/Migration_71.kt:9]: Will do that from next migrations 👍🏾

File: app/src/main/java/org/simple/clinic/login/pin/LoginPinEffectHandler.kt (hunk 3 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/login/pin/LoginPinEffectHandler.kt:92]: Ah okay, my bad. I was under the impression that `OngoingLoginEntry` is only local to device for registration purpose.

File: app/src/main/java/org/simple/clinic/user/LoggedInUserPayload.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/user/LoggedInUserPayload.kt:36]: Hmm, alright. Let's add another migration for that once it's finalized.

**Comment by vinaysshenoy** [comment]
@msasikanth Do we need to update the CHANGELOG here? We have updated the database schema, so I think adding a line is warranted.

**Comment by msasikanth** [comment]
> @msasikanth Do we need to update the CHANGELOG here? We have updated the database schema, so I think adding a line is warranted.

Yeah, I was thinking about having a in-progress feature were we track all these changes. But I guess I can place this change in internal.

**Review by vinaysshenoy — APPROVED** [review]

**Review by vinaysshenoy — APPROVED** [review]
